managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3117,8 +3125,11 @@ public void offloadFailed(ManagedLedgerException e, Object ctx) {
 
     @Override
     public void asyncOffloadPrefix(Position pos, OffloadCallback callback, Object ctx) {
-        if (config.getLedgerOffloader() != null && config.getLedgerOffloader() == NullLedgerOffloader.INSTANCE) {
-            callback.offloadFailed(new ManagedLedgerException("NullLedgerOffloader"), ctx);
+        LedgerOffloader ledgerOffloader = config.getLedgerOffloader();
+        if (ledgerOffloader != null && (ledgerOffloader == NullLedgerOffloader.INSTANCE
+                || ledgerOffloader instanceof ReadonlyWrapperLedgerOffloader)) {
```

Comment:
Instead of adding a `instanceof` check, would it make sense to add a new default method `isAppendable()` to the LedgerOffloader interface? Based on the comment in https://github.com/apache/pulsar/pull/23279/files#r1751500211, it seems that `Readonly` is not the correct concept.

Reply:
Thanks for your suggestion, it is very useful.  Fixed